pkg/client/fake/client.go

```diff
@@ -79,28 +81,42 @@ func NewFakeClientWithScheme(clientScheme *runtime.Scheme, initObjs ...runtime.O
 }
 
 func (t versionedTracker) Create(gvr schema.GroupVersionResource, obj runtime.Object, ns string) error {
-	if accessor, err := meta.Accessor(obj); err == nil {
-		if accessor.GetResourceVersion() == "" {
-			accessor.SetResourceVersion("1")
-		}
-	} else {
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
 		return err
 	}
+	if accessor.GetResourceVersion() != "" {
+		return apierrors.NewBadRequest("resourceVersion can not be set for Create requests")
+	}
+	accessor.SetResourceVersion("1")
 	return t.ObjectTracker.Create(gvr, obj, ns)
 }
 
 func (t versionedTracker) Update(gvr schema.GroupVersionResource, obj runtime.Object, ns string) error {
-	if accessor, err := meta.Accessor(obj); err == nil {
-		version := 0
-		if rv := accessor.GetResourceVersion(); rv != "" {
-			version, err = strconv.Atoi(rv)
-		}
-		if err == nil {
-			accessor.SetResourceVersion(strconv.Itoa(version + 1))
-		}
-	} else {
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		return fmt.Errorf("failed to get accessor for object: %v", err)
+	}
+	oldObject, err := t.ObjectTracker.Get(gvr, ns, accessor.GetName())
+	if err != nil {
+		return err
+	}
+	oldAccessor, err := meta.Accessor(oldObject)
+	if err != nil {
 		return err
 	}
+	if accessor.GetResourceVersion() != oldAccessor.GetResourceVersion() {
+		return apierrors.NewConflict(gvr.GroupResource(), accessor.GetName(), errors.New("object was modified"))
```

Comment:
`NewResourceExpired` may be the right one.
https://godoc.org/k8s.io/apimachinery/pkg/api/errors#NewResourceExpired

Reply:
I don't think so, because this is the error message you see on conflicts: https://github.com/kubernetes/apimachinery/blob/6584f51ae935c6ffdc5abd5fc5cd1e640aab8a7f/pkg/api/errors/errors.go#L207